core/src/main/scala/org/http4s/headers/Content-Length.scala

```diff
@@ -12,18 +12,24 @@ import org.http4s.util.Writer
   *
   * @param length the length; throws an `IllegalArgumentException` if negative
   */
-final case class `Content-Length`(length: Long) extends Header.Parsed {
-  require(length >= 0L)
-
+sealed abstract case class `Content-Length`(length: Long) extends Header.Parsed {
```

Comment:
I know I'm a bit late to this party, but I'm not a huge fan of using a sealed abstract case class with a bunch of anonymous instances: `zero` is a unique class, the result of `modify` is a unique class, etc. This isn't a problem if you either funnel all calls through one constructor or, IMO better, make a 
```scala
private class ContentLenthImpl(l: Length) extends `Content-Length`(l)`
```
in the object to use.